### PR TITLE
Update CBOR to always have an accept header

### PR DESCRIPTION
**Pull request by mullermp**
Needs to update Smithy and test against protocol tests.

### CODE REVIEW

**Review by alextwoods — APPROVED** [review]

**Review by alextwoods — APPROVED** [review]
Looks good.

Worth adding a changelog entry for?

File: codegen/smithy-ruby-codegen/CHANGELOG.md [hunk 1]
```diff
@@ -1,6 +1,8 @@
 Unreleased Changes
 ------------------
 
+* Issue - Support services migrating from AWS Query to AWS JSON or RPC v2.
```
Comment (alextwoods) [inline codegen/smithy-ruby-codegen/CHANGELOG.md:4]: This AWS specific bit is kind of weird to have in our generic changelog.

File: codegen/smithy-ruby-codegen/src/main/java/software/amazon/smithy/ruby/codegen/generators/ParserGeneratorBase.java [hunk 2]
```diff
@@ -301,9 +300,9 @@ public void render(FileManifest fileManifest) {
     }
 
     protected void renderParsers() {
-        TreeSet<Shape> shapesToRender = CodegenUtils.getAlphabeticalOrderedShapesSet();
-        TreeSet<Shape> eventStreamEventsToRender = CodegenUtils.getAlphabeticalOrderedShapesSet();
-
+        Comparator<Shape> comparator = Comparator.comparing(o -> o.getId().getName() + " " + o.getId());
```
Comment (alextwoods) [inline codegen/smithy-ruby-codegen/src/main/java/software/amazon/smithy/ruby/codegen/generators/ParserGeneratorBase.java:303]: Why this change? The codegenutils helper method for this seems cleaner than implementing a comparator for each one?

**Review by alextwoods — APPROVED** [review]

File: codegen/smithy-ruby-codegen/CHANGELOG.md (hunk 1 above)
Comment (mullermp) [reply codegen/smithy-ruby-codegen/CHANGELOG.md:4]: Yeah. I can change this "from AWS Query to RPC v2" - the query trait (AWS) is used generically now so that bit must stay.
